### PR TITLE
vagrantfile: Fix vagrant up fedora

### LINKED ISSUE #3760: Vagrant fedora machine does not come up
# Description of problem

With the draft pull request I was able to bring up a Fedora based VM for running CI tests using `vagrant up fedora`.  However with the merged version of the vagrant patches, attempting to bring up the Fedora VM no longer works (the Ubuntu based one seems to get further).

# Expected result

```
$ vagrant up fedora
$ vagrant ssh fedora
```

Can then use the fedora VM.

# Actual result

```
$ vagrant up fedora
[ . . . ]
$ vagrant ssh fedora
==> fedora: Running action triggers before ssh ...
==> fedora: Running trigger...
==> fedora: Domain is not created. Please run `vagrant up` first.
```

# Further information

Full transcript to follow.

### PR DESCRIPTION
**Pull request by wainersm**
The aim with this pull request is to fix the `vagrant up fedora` command which is not currently working. I also send a bonus commit that is just an improvement. I tested both `vagrant up fedora` and `vagrant up ubuntu` to ensure no regressions with ubuntu.

@dgibson reported the problem on #3760 but actually I never hit the same issue. Here instead it has crashed because of commit 26cef9d.

@Jakob-Naucke I didn't touch your code which forces the use of Podman on Fedora-likes. Instead I ensure it work with this vagrant file. TBH, I don't know why (and didn't investigate) on the Fedora box with cgroups v2, Podman fails to build the guest image.

Fixes #3760

### CODE REVIEW

**Review by dgibson — APPROVED** [review]
It took me a bunch of attempts to get this working... but that appears to be because one of the Fedora servers was flaky, not anything to do with the patches.

Seems to work, excellent!

File: .ci/setup_env_ubuntu.sh [hunk 1]
```diff
@@ -102,6 +102,12 @@ main()
 		pkgs_to_install+=" ${rust_agent_pkgs[@]}"
 	fi
 
+	# The redis-server package fails to install if IPv6 is disabled. Let's
+	# check if that's the case and then enable it.
+	if [[ $(sudo sysctl -n net.ipv6.conf.all.disable_ipv6) -eq 1 ]]; then
```
Comment (jodh-intel) [inline .ci/setup_env_ubuntu.sh:107]: Nit: You don't need to use double square brackets here.

**Review by jodh-intel — APPROVED** [review]
Thanks @wainersm.

**Review by Jakob-Naucke — COMMENTED** [review]
~Hmm, I still get the `sd-bus call: Transport endpoint not connected` with Fedora.~ (see below) Btw, just recently I found out you _can_ get Docker on the IBM architectures, but apparently not on RHEL (which I would have to use for CI).

File: .ci/setup_env_ubuntu.sh (hunk 1 above)
Comment (wainersm) [reply .ci/setup_env_ubuntu.sh]: @jodh-intel just updated it to remove the extra square bracket. Thanks!

**Comment by wainersm** [comment]
> Hmm, I still get the `sd-bus call: Transport endpoint not connected` with Fedora. Btw, just recently I found out you _can_ get Docker on the IBM architectures, but apparently not on RHEL (which I would have to use for CI).

Double-checking: using Vagrant you are still getting the sd-bus error? Did you `destroy`  then `up` the vm?

**Comment by Jakob-Naucke** [comment]
@wainersm scratch that -- I had forgotten to apply your changes :man_facepalming: taking a look...

**Comment by Jakob-Naucke** [comment]
Did anyone else observe this?
```
Vagrant attempted to execute the capability 'reboot'
on the detect guest OS 'fedora', but the guest doesn't
support that capability. This capability is required for your
configuration of Vagrant. Please either reconfigure Vagrant to
avoid this capability or fix the issue by creating the capability.
```

This is system-installed Vagrant 2.2.6 on Ubuntu 20.04 x86_64 with libvirt.

**Comment by wainersm** [comment]
> Did anyone else observe this?
> 
> ```
> Vagrant attempted to execute the capability 'reboot'
> on the detect guest OS 'fedora', but the guest doesn't
> support that capability. This capability is required for your
> configuration of Vagrant. Please either reconfigure Vagrant to
> avoid this capability or fix the issue by creating the capability.
> ```
> 
> This is system-installed Vagrant 2.2.6 on Ubuntu 20.04 x86_64 with libvirt.

I always tested the vagrantfile on my Fedora 33 x86_64 machine. Let me create a Ubuntu VM to give it a try.
thanks for reporting it @Jakob-Naucke !

**Review by GabyCT — APPROVED** [review]

**Comment by wainersm** [comment]
> > Did anyone else observe this?
> > ```
> > Vagrant attempted to execute the capability 'reboot'
> > on the detect guest OS 'fedora', but the guest doesn't
> > support that capability. This capability is required for your
> > configuration of Vagrant. Please either reconfigure Vagrant to
> > avoid this capability or fix the issue by creating the capability.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This is system-installed Vagrant 2.2.6 on Ubuntu 20.04 x86_64 with libvirt.
> 
> I always tested the vagrantfile on my Fedora 33 x86_64 machine. Let me create a Ubuntu VM to give it a try.
> thanks for reporting it @Jakob-Naucke !

@Jakob-Naucke I could reproduce it in a ubuntu 20.04 VM. I will open another issue. are you okay if I merge this fix?

**Pull request merged**